app/messaging/process-message.js

```diff
@@ -0,0 +1,58 @@
+const util = require('util')
+const { set, update } = require('../repositories/application-repository')
+const { applicationResponseMsgType, applicationResponseQueue } = require('../config')
+const sendMessage = require('../messaging/send-message')
+const createReference = require('../lib/create-reference')
+const { notify: { templateIdApplicationComplete } } = require('../config')
+const sendEmail = require('../lib/send-email')
+const processApplication = async (msg) => {
+  const responseMessage = msg.body
+  let reference = ''
+  try {
+    // Get ID
+    const result = await set({
+      reference,
+      data: JSON.stringify(msg.body),
+      createdBy: 'admin',
+      createdAt: new Date()
+    })
+    // GetReference for ID
+    const application = result.dataValues
+    reference = msg.body.applicationId = createReference(application.id)
+    // Update
+    await update({
```

Comment:
This update could be replaced by adding a hook to the `application` model e.g.
```js
hooks: {
  afterCreate: async (application, options) => {
    application.dataValues.reference = createReference(application.id)
    await application.update(application.dataValues)
  }
}
```
I think using the hooks option is a bit safer and ensures the record is created with the reference value without having to remember to do it when creating the record in code. What do you think?